aws/sdk-codegen/src/test/kotlin/software/amazon/smithy/rustsdk/SigV4AuthDecoratorTest.kt

```diff
@@ -56,10 +62,113 @@ class SigV4AuthDecoratorTest {
+                                    "name": "sigv4a",
+                                    "signingRegionSet": ["*"],
+                                    "signingName": "dontcare"
+                                }
+                            ]
+                        }
+                    }
+                }
+            ],
+            "parameters": {
+                "endpoint": { "required": true, "type": "string", "builtIn": "SDK::Endpoint" },
+            }
+        })
+        @service(sdkId: "dontcare")
+        service TestService { version: "2023-01-01", operations: [SomeOperation] }
+
+        @streaming
+        blob Bytestream
+
+        structure SomeInput {
+            @httpPayload
+            @required
+            something: Bytestream
+         }
+
+        structure SomeOutput { something: String }
+
+        @http(uri: "/", method: "POST")
+        operation SomeOperation { input: SomeInput, output: SomeOutput }
+        """.asSmithyModel(smithyVersion = "2.0")
+
+    @Test
+    fun unsignedPayloadSetsCorrectHeaderForSigV4a() {
+        awsSdkIntegrationTest(modelWithSigV4aAuthScheme) { clientCodegenContext, rustCrate ->
+            val moduleUseName = clientCodegenContext.moduleUseName()
+            val rc = clientCodegenContext.runtimeConfig
+
+            rustCrate.integrationTest("sigv4a") {
+                Attribute.featureGate("test-util").render(this)
+                tokioTest("test_sigv4a_signing") {
+                    rustTemplate(
+                        """
+                        let http_client = #{StaticReplayClient}::new(vec![#{ReplayEvent}::new(
+                            #{Request}::builder()
+                                .header("authorization", "AWS4-ECDSA-P256-SHA256 Credential=ANOTREAL/20090213/dontcare/aws4_request, SignedHeaders=content-length;content-type;host;x-amz-date;x-amz-region-set;x-amz-user-agent, Signature=3045022100b95d1c054ff04b676d12f0c893348606844d67ccf595981f0ca4968fae2eddfd022073e66edc0ad1da05b08392fccefa3ad69f8ec9393461033412fa05c55b749e9d")
+                                .uri("https://example.com")
+                                .body(#{SdkBody}::from("Hello, world!"))
+                                .unwrap(),
+                            #{Response}::builder().status(200).body(#{SdkBody}::empty()).unwrap(),
+                        )]);
+                        let config = $moduleUseName::Config::builder()
+                            .http_client(http_client.clone())
+                            .endpoint_url("https://example.com")
+                            .behavior_version_latest()
+                            .with_test_defaults()
+                            .build();
+                        let client = $moduleUseName::Client::from_conf(config);
+                        let _ = client.some_operation().something(#{ByteStream}::from_static(b"Hello, world!")).send().await;
+
+                        http_client.assert_requests_match(&[]);
```

Comment:
A bit concerned the use of `assert_requests_match` since the verification tries matching `Signature`, which is subject to change as we add new business metrics moving forward (new business metrics may be included in `x-amz-user-agent` by default in SDK codegen decorators, so it can affect a Kotlin test like this one). When new business metrics were introduced in a PR, that PR would probably fail this test as part of semver hazards checks (which is why we introduced `relaxed_validate` in [this PR](https://github.com/smithy-lang/smithy-rs/pull/3786) to work around the problem).

Long story short, to verify the expected output of the test, I'd probably get the `authorization` header and check if it contains `AWS4-ECDSA-P256-SHA256`.

Reply:
OK, sounds good.